### PR TITLE
Added command extension point draft

### PR DESCRIPTION
**Pull request by norberttech**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| License | MIT |

### CODE REVIEW

File: src/Dumplie/Application/Command/Extension.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+use Dumplie\Application\ServiceLocator;
+
+interface Extension
+{
+    /**
+     * @param Command $command
+     * @return bool
+     */
+    public function expands(Command $command) : bool;
```
Comment (potfur) [inline src/Dumplie/Application/Command/Extension.php:15]: :+1:

File: src/Dumplie/Application/Command/Extension.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    /**
+     * @param Command $command
+     * @param ServiceLocator $serviceLocator
+     */
+    public function pre(Command $command, ServiceLocator $serviceLocator);
```
Comment (potfur) [inline src/Dumplie/Application/Command/Extension.php:21]: In some cases `pre` and `post` hooks would need to be in same transaction as command handler.

Comment (norberttech) [inline src/Dumplie/Application/Command/Extension.php:21]: yea, good point. Im going to work on that

**Comment by norberttech** [comment]
@potfur pointed out that some extensions might need to be handled in the same transaction that we use to handle commands. For example if we decide to create extension that will update metadata for orders we need to be sure that those operations (command and extension) are atomic. 
Basically I added possibility to execute extensions for commands in specific order. Each extension can be registered with priority. Moving transactions into extensions created a possibility to register extensions that will be handled in the same transactions that is used to handle command itself. 

I also decided to remove specs for inventory command handlers, they were inconsistent with other commands that we have.

**Pull request merged**